src/collections/vector.js

```diff
@@ -0,0 +1,128 @@
+import * as near from '../api'
+import {u8ArrayToString} from '../utils'
+
+const ERR_INDEX_OUT_OF_BOUNDS = "Index out of bounds"
+const ERR_INCONSISTENT_STATE = "The collection is an inconsistent state. Did previous smart contract execution terminate unexpectedly?"
+
+/// An iterable implementation of vector that stores its content on the trie.
+/// Uses the following map: index -> element
+export class Vector {
+    constructor(prefix) {
+        this.length = 0
+        this.prefix = prefix
+    }
+
+    len() {
+        return this.length
+    }
+
+    isEmpty() {
+        return this.length == 0
+    }
+
+    indexToKey(index) {
+        // TODO: there's no efficient way to convert a Uint64 (BigInt) to Uint8Array
+        // we limit index to be Uint32 for now
```

Comment:
I was referencing to old collections/Vector instead of new store/vec, which is using u64. Because examples are using `collections/*` and we're trying to achieve features parallel to near-sdk-rs 1.0 or 2.0. Since new one has moved to u32, it's good now!